components/raftstore-v2/src/raft/storage.rs

```diff
@@ -41,19 +43,102 @@ pub fn write_initial_states(wb: &mut impl RaftLogBatch, region: Region) -> Resul
 }
 
 /// A storage for raft.
+#[derive(Debug)]
 pub struct Storage<ER> {
     engine: ER,
+    peer: metapb::Peer,
+    region_state: RegionLocalState,
+    raft_state: RaftLocalState,
+    apply_state: RaftApplyState,
     logger: Logger,
 }
 
-impl<ER> Storage<ER> {
-    pub fn new(engine: ER, logger: Logger) -> Storage<ER> {
-        Storage { engine, logger }
+impl<ER: RaftEngine> Storage<ER> {
+    /// Creates a new storage.
+    ///
+    /// All metadata should be initialized before calling this method. If the region is destroyed
+    /// `None` will be returned.
+    pub fn new(
+        engine: ER,
+        region_id: u64,
+        store_id: u64,
+        logger: &Logger,
+    ) -> Result<Option<Storage<ER>>> {
```

Comment:
suggestion: a more natural way of organize the parameters order is "store_id, region_id, engine, logger".

Reply:
We usually put `region_id` first.